hnn_core/network_builder.py

```diff
@@ -376,30 +373,21 @@ def _create_cells_and_drives(self, threshold, record_vsoma=False,
         These drives are spike SOURCES but cells are also targets.
         External inputs are not targets.
         """
-        type2class = {'L2_pyramidal': pyramidal, 'L5_pyramidal': pyramidal,
-                      'L2_basket': basket, 'L5_basket': basket}
+        self.net._update_cells()
         # loop through ALL gids
         # have to loop over self._gid_list, since this is what we got
         # on this rank (MPI)
-
         for gid in self._gid_list:
-            src_type, src_pos, is_cell = self.net._get_src_type_and_pos(gid)
```

Comment:
if you're removing the call to this function, shouldn't you remove the function as well?

Reply:
Good catch! Addressed in https://github.com/jonescompneurolab/hnn-core/pull/321/commits/e9c79bf5cb3f9047ec463f00e9d9986965034121.